### PR TITLE
Add Google Auth

### PR DESCRIPTION
**Pull request by mpah**
Tested. Problem before was not having it in heartbeat as well.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,16 @@ requests==2.10.0
 flask-googlemaps==0.2.0
 s2sphere==0.2.4
 requests==2.10.0
```
Comment (marauder37) [inline requirements.txt]: requests is in here twice, is that an old regression or something new?

**Comment by mpah** [comment]
Ah should only be once and should be fixed but pip will just skip it (cached). Im kinda new to this whole pull request thing, so should I fix it and redo the pull request?

**Comment by sykhro** [comment]
just fix, commit, and rebase. then someone will merge

**Comment by marauder37** [comment]
I'm trying to test this but having a bugger of a time installing `cryptography` on OSX

**Comment by mpah** [comment]
@sykhro Ok I have commited the fix, how do I rebase?

**Comment by sykhro** [comment]
you need to 
`git pull
git rebase
git commit
git push`

**Comment by mpah** [comment]
Oh its in there already, I committed and pushed.

**Comment by sykhro** [comment]
you need to do it again after pulling and rebasing, doesn't seem like the latest changes are in your pull request

**Comment by marauder37** [comment]
Don't worry about rebasing that last commit, having 3 commits in the PR won't kill us. But you will have to rebase against `AHAAAAAAA/master`, we can guide you through it once I've seen more of your code

**Comment by mpah** [comment]
Isnt it already based against AHAAAAAAA/master?

**Comment by marauder37** [comment]
Nah, see: https://snag.gy/JArbqZ.jpg . The most recent changes are at the top, and the tip of your mpah/master branch is about 20 commits behind. We'll bring it up to date

**Comment by marauder37** [comment]
What tool are you using to work with git?

**Comment by mpah** [comment]
Im using sourcetree, so anything I should do?

**Comment by marauder37** [comment]
Do you have a remote for the main repo? In that screenshot mine is `aha`

**Comment by mpah** [comment]
the remote is my own forked from aha

**Comment by marauder37** [comment]
ok so in the Repository menu go Add remote and call it aha and URL https://github.com/AHAAAAAAA/PokemonGo-Map.git

**Comment by mpah** [comment]
ok what next?

**Comment by marauder37** [comment]
image

Hopefully your thing looks like that, and you can see your local master, aha/master, and mpah/master which is on github

Dblclick local master to make sure it's checked out

Right click aha/master and choose rebase onto master

Git will do its best to drag your history forwards automatically, but in this case it will make a mess

First clean up requirements.txt, that's easy. Click Working Copy then unstaged files > requirements.txt

the bit between +<<<<<<< HEAD and +======= is the code in the main repo and the bit between +======= is the code in your repo that couldn't be straightened out

Edit the file, get rid of all ===== markings etc and make sure each requirement is listed once, then add requiprements.txt to the staged files area

**Comment by mpah** [comment]
Hmm do you mean Pull into current branch?
image

**Comment by marauder37** [comment]
image

**Comment by mpah** [comment]
Sorry for being a noob, I tried to do it via command line
image
where do I go now to fix the conflicts manually?

**Comment by marauder37** [comment]
My first taste of google auth is so good that all is forgiven... 

Open up requirements.txt in your text editor and straighten it out as mentioned earlier, then git add it

**Comment by mpah** [comment]
Isnt that already done in https://github.com/AHAAAAAAA/PokemonGo-Map/pull/24/commits/0ccdb4a47471160583865bf8984a888a83410f95?

**Comment by marauder37** [comment]
There is no 'already done' for resolving merge conflicts. You just have to fix what's in front of you.  The past is the past :)  In this case you fixed it but the fix moved something else so the two versions won't slide into each other neatly.

**Comment by sykhro** [comment]
I have merged this manually and tagged you in the commit. Thanks for your contribution.

**Comment by sykhro** [comment]
welp, something went wrong. please try again

**Comment by mpah** [comment]
Oh I was about to commit my merge, I've used the tortoise mergetool. Thanks for the patience, git still has me confused a lot.

**Comment by mpah** [comment]
Should I push my merge?

**Comment by sykhro** [comment]
Please do. For some reason what I did didn't work.

**Comment by MonkeyLeeT** [comment]
Thanks for the awesome pull. Are you also going to modify your code so that ptc/google login coexist depending on another option flag?

**Comment by marauder37** [comment]
@monkeyleet there's already a -a [google|ptc] flag for that

**Comment by mpah** [comment]
Hmm it shows me 3 new commits to pull hold on

**Comment by sykhro** [comment]
I forked this, rebased manually, and published. For reasons unknown it produced an empty commit (c23530d)

**Comment by mpah** [comment]
For some reason I now have .orig files ugh..
How about I just get the latest version and manually readd the changes, I probably just need 5 min?
So just dont commit anything meanwhile

**Comment by sykhro** [comment]
I'm tempted to try again and amend but I would probably end up making a mess. At this point reading the changes manually would be the safest thing to do

**Pull request closed without merging**